### ISSUE TITLE
git hub pull-request fails on Windows

### ISSUE THREAD

**Issue by bendtherules** [seveas/git-spindle#103]
Command:
`git hub pull-request` on Windows from another branch (to master)

I am guessing the unix-ness of the test suite means it is usually not tested on Windows

Traceback:

    Traceback (most recent call last):
      File "C:\Python27\Scripts\git-hub", line 11, in <module>
        GitHub().main()
      File "C:\Python27\lib\site-packages\gitspindle\__init__.py", line
    309, in main
        func(opts)
      File "C:\Python27\lib\site-packages\gitspindle\github.py", line 11
    20, in pull_request
        self.gitm('fetch', remote, redirect=False)
      File "C:\Python27\lib\site-packages\gitspindle\__init__.py", line
    102, in gitm
        result = self.git(*args, **kwargs)
      File "C:\Python27\lib\site-packages\whelk\__init__.py", line 144,
    in __call__
        if sys.platform != 'win32' or (kwargs['stdin'], kwargs['stdout']
    , kwargs['stderr']).count(None) == 3:
    KeyError: 'stdin'

**Comment by seveas** [comment]
That's actually a bug in whelk that I happened to find a few days ago. I'm not a windows user though, so have a problem testing the fix.

Could you install whelk from https://seveas.net/whelk-2.5.2.tar.gz and try again? There's also a python 3.5-specific fix in there, if you happen to have 3.5 installed as well I'd appreciate a test with it.

**Comment by seveas** [comment]
Nad about the testsuite: I'm afraid you are correct. The testsuite is a fairly recent thing, and I'm trying to make it more complete before porting it to work on windows :)

Please don't let this discourage you from reporting issues though, git-spindle should work just fine on windows and I'll do my best to make it so.

**Comment by bendtherules** [comment]
So, whelk is like a dependency of git-spindle, right?

**Comment by bendtherules** [comment]
Update:

I finally managed to get it working, by including the wordpad path to env. I think this was very much a corner case as apparently `cmd` on windows doesnt know the path of wordpad automatically, which is where `shell=true` redirects, but `powershell` knows it automatically even when it is not in the env path.

So, not at all a mistake from `git-spindle`.

**But I think there is another interesting thing too:**

I saw that `_getitem` tries all env path and also append `.exe` to search for the file on Windows, but when I used just `editor=wordpad` i.e. without the `.exe`, it never actually reached that function. `_call_` tries to resolve it via popen and fails.
